### DISCUSSION TITLE
Some more detailed answers to questions about Lenovo Yoga Slim 7x hardware support

### DISCUSSION THREAD

**Discussion by kuruczgy** [kuruczgy/x1e-nixos-config#20] [General]
Someone who is researching what new laptop to buy recently asked a bunch of questions over email, we thought the answers could be useful to other people as well, so I am posting them here:

> 1. The README says there is GPU acceleration support. Does it also mean GPU video decoding and encoding working?

Firefox about:support reports that only software decoding is supported for everything. Not sure how difficult it would be to get it working, for me it's fairly low priority as the CPU seems to be plenty fast enough for SW decoding. (The only time the laptop gets uncomfortably hot is when kernel compiling, watching movies and youtube videos only gets it lukewarm.)

> 2. Are there communities of people gathering and making patches for the X Elite laptops? Where can I find patches for the laptop and see status of certain features?

The main community hub seems to be the #aarch64-laptops IRC channel on OFTC, I would definitely recommend you to join there. (You can also join using the Matrix bridge.) Unfortunately there is no community wiki (I guess nobody bothered to set one up yet), so most knowledge is unwritten/ephemeral.

You could ask around about other X1E laptops on IRC as well, I think there is at least one owner of almost every laptop in the channel.

> 3. Which model of the Lenovo Yoga Slim 7x do you have and is supported by the repository? I saw you can get it with 16/32 GB RAM and 512GB/1TB storage, and I was wondering what you have since I know these things already work for yours.

I have the 32GB/1TB model. There is a small risk that RAM size difference matters, some people had issues with the 64GB version of the thinkpad T14s not booting and needing some patches compared to the 32GB version. I cannot possibly imagine NVMe size mattering for compatibility. (Many people are also swapping their NVMe drives, I haven't heard about issues yet.)

> 4. I've seen in the README that sleep and speakers aren't working. What other features aren't working yet(For Lenovo Yoga Slim 7x)?

Sleep actually _is_ working, it just isn't a very deep sleep yet (consumes about 3.8%/hour battery according to my measurements), and it sometimes wakes up spuriously. But I do use it sometimes, it's still better than nothing.

An annoying issue right now that's specific to the yoga slim 7x is that we still weren't able to get the lid switch working. (AFAIK it works on most other X1E laptops, but there is something funny going on with the yoga 7x.) So the sleep process right now looks like this:
- Send the laptop to sleep manually (type `systemctl suspend`, or set up a hotkey for it)
- Close the lid
- Open the lid
- Press a key to wake the laptop

> That might be too broad to ask, so do these features I listed work for you? And if not, do you know what is their status? (you never tired, never worked on, in progress by someone, heard it works for someone, patches exists but not working for you, etc...)
> - Touchpad
> - Built in keyboard
> - Hibernate (I assume no since sleep isn't working)
> - Swap memory
> - Camera
> - AUX port
> - USB ports
> - External displays and display over USB-C
> - Keyboard backlight
> - Sensors

- Touchpad: working from day 1 without issues (even better than on windows, windows seems to be sometimes not registering clicks)
- Keyboard: for a few days it was not working in GRUB, but I think a firmware update or something solved it, has been working flawlessly since
- Hibernate: never tried, but someone on IRC has been playing around with it
- Swap memory: never tried, but I don't see any reason it wouldn't work
- Camera: never tried, and I don't see any device tree bindings, so it will probably take a while until someone gets it working
- AUX port: if you mean the one on the dongle (comes with the laptop), I just tried, it works. Remember that this laptop has _only_ USB-C ports.
- USB ports: haven't had issues. (Funnily USB-A support is somewhat more troublesome at least according to the discussions I saw on IRC, fortunately this laptop does not have USB-A ports 🙂 )
- USB-C DP alt mode: never tried, but there is some very active work going on based on the discussions I saw on IRC. From what I gather the status is "it kinda works but there are still glitches" (e.g. stuff like it only working in one orientation of the USB-C plug)
- Keyboard backlight: not controlled by the OS, works just fine. (And a patch yesterday just dropped to automatically make it turn off during sleep, I will soon add that to the repo as well.)
- Sensors: never tried, I think the only "sensor" the laptop has is the facial recognition IR camera? (Or if you mean thermal sensors those work fine, according to sysfs there are 50+ thermal zones on this laptop and they all seem to be working correctly.)

> 5. How is the battery times and power efficiency? I know Linux can be problematic with this.

Feels solid to me, e.g. I can easily do 8+ hour coding sessions on it without needing to plug it in. In general linux support feels really solid while the laptop is running, and I don't know of anything that linux doesn't support yet that would make it less power efficient than windows. (But I never did any measurements, so who knows.)

**Comment by alyraffauf** [comment]
Is it possible to come up with a feature matrix not unlike the ones that asahi has? Their SoC support matrix is a little too detailed, but we could devise a practical one (wifi, bluetooth, suspend, speakers, etc).
![image](https://github.com/user-attachments/assets/8ccd5794-553f-4458-a9cf-5217b9bbf917)
![image](https://github.com/user-attachments/assets/57af0cd6-907b-4a6a-bba8-c5fdb47656c0)
